[FIX] Hugely degraded performance for the translation subquery inject…

eantones opened the pull request:
…ion in _generate_translated_field https://github.com/odoo/odoo/issues/18679

It solves issue https://github.com/odoo/odoo/issues/18679 (https://github.com/odoo/odoo/issues/18679) on Odoo v10

Only works on PostgreSQL 9.5 or above

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

eantones committed:
[FIX] Hugely degraded performance for the translation subquery injection in _generate_translated_field #18679

It solves issue #18679 (https://github.com/odoo/odoo/issues/18679) on Odoo v10

Only works on PostgreSQL 9.5 or above

FFernandez-PlanetaTIC committed:
[FIX] base: create index as unique and remove duplicates before

FFernandez-PlanetaTIC commented:
Hi @eantones,

Will you propose this PR to odoo/OCA repository instead of yours?

If not we can do it.

Regards

eantones commented:
Hi @FFernandez-PlanetaTIC ,

you mean to propose to the Odoo official repo, right?

I was going to do that a long time ago but after reading some comments on https://github.com/odoo/odoo/issues/18679 like:

https://github.com/odoo/odoo/issues/18679#issuecomment-324800841

https://github.com/odoo/odoo/issues/18679#issuecomment-354811670

https://github.com/odoo/odoo/issues/18679#issuecomment-355966533

I understood that they're never going to merge it, that's why I asked politely in https://github.com/odoo/odoo/issues/18679#issuecomment-400103756. But we can try now, at least to have a PR in the official Odoo repo to merge directly on our repos or using tools like git-aggregator. What do you think?

Thanks!

FFernandez-PlanetaTIC commented:
I think in OCA/OCB is more likely to be accepted.

eantones commented:
Ok, let's see what OCA/OCB says :-)

https://github.com/OCA/OCB/pull/899

eantones committed:
[FIX] base: Make it compatible with all versions of PostgreSQL supported by Odoo 10.0

eantones committed:
[FIX] base: Fix issues importing translations

Based on https://github.com/PlanetaTIC/odoo/commit/3b3c9c687d013200db49484d2fbb4a8b45f7a52d correction.

- Previous commit breaks import (from ONLY needed in where not exists)
- Avoid duplicates in import (a slightly clearer and cleaner approach)
